e2e/tools/validator/src/validator/prometheus/__init__.py

```diff
@@ -1,66 +1,127 @@
-                    cleaned_validated_data.append(float(element[1]))
-                else:
-
-                    cleaned_validated_data[index] += float(element[1])
-        return cleaned_validator_data, cleaned_validated_data
-
-
-def deltas_func(validator_data, validated_data) -> List[float]:
-    delta_list = []
-    for validator_element, validated_element in zip(validator_data, validated_data):
-        delta_list.append(abs(validator_element - validated_element))
-    return delta_list
-
-def percentage_err(validator_data, validated_data) -> List[float]:
-    percentage_err_list = []
-    for validator_element, validated_element in zip(validator_data, validated_data):
-        percentage_err_list.append(abs((validator_element - validated_element) / validator_element) * 100)
-    return percentage_err_list
+def disjunct_on_timestamps(prom_data_list_one, prom_data_list_two) -> Tuple[list, list]:
+    common_timestamps = [int(datapoint[0]) for datapoint in prom_data_list_one if int(datapoint[0]) in [int(datapoint[0]) for datapoint in prom_data_list_two]]
+    # necessary to sort timestamps?
+    common_timestamps.sort()
+    list_one_metrics = []
+    list_two_metrics = []
+    for timestamp in common_timestamps:
+        for list_one_datapoint in prom_data_list_one:
+            if int(list_one_datapoint[0]) == timestamp:
+                list_one_metrics.append(list_one_datapoint[1])
+        for list_two_datapoint in prom_data_list_two:
+            if int(list_two_datapoint[0]) == timestamp:
+                list_two_metrics.append(list_two_datapoint[1])
+    return list_one_metrics, list_two_metrics
 
+
+def compare_metrics(endpoint: str, disable_ssl, start_time: datetime, end_time: datetime, expected_query: str, expected_query_labels: dict, actual_query: str, actual_query_labels: dict) -> Tuple[List[float], List[float]]:   
+    prom_client = PrometheusConnect(endpoint, headers=None, disable_ssl=disable_ssl)
+
+    expected_metrics = prom_client.get_metric_range_data(
+        metric_name=expected_query,
+        label_config=expected_query_labels.copy(),
+        start_time=start_time,
+        end_time=end_time,
+        params={
+            "step": "3s"
+        }
+
+    )
+    actual_metrics = prom_client.get_metric_range_data(
+        #metric_name=actual_query,
+        metric_name="kepler_node_platform_joules_total{job='vm'}",
+        #label_config=actual_query_labels.copy(),
+        start_time=start_time,
+        end_time=end_time,
+        params={
+            "step": "3s"
+        }
+    )
+    print(expected_metrics)
+    print(actual_metrics)
+    # clean data to acquire only lists
+    expected_data = merge_prom_metric_list(expected_metrics)
+    actual_data = merge_prom_metric_list(actual_metrics)
+    
+    # remove timestamps that do not match
+    return disjunct_on_timestamps(expected_data, actual_data)
+
+
+def absolute_percentage_error(expected_data, actual_data) -> List[float]:
```

Comment:
I feel these pure functions can be easily unit-tested. Could you please add to the existing test suite?